src/Sarif/Writers/MimeType.cs

```diff
@@ -39,6 +34,12 @@ public static string DetermineFromFileExtension(string path)
                     }
                 }
             }
+
+            if (System.IO.Directory.Exists(path))
```

Comment:
Great! This should help quite a bit by itself.

Did you also defer the check in FileData.Create when we didn't request any dataToInsert? That was the other half of the performance issue.

Reply:
Cool, that works. =)